x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/Automatons.java

```diff
@@ -69,11 +95,23 @@ public static Automaton patterns(Collection<String> patterns) {
      * Builds and returns an automaton that represents the given pattern.
      */
     static Automaton pattern(String pattern) {
+        if (cache == null) {
+            return buildAutomaton(pattern);
+        } else {
+            try {
+                return cache.computeIfAbsent(pattern, ignore -> buildAutomaton(pattern));
+            } catch (ExecutionException e) {
+                return handleCacheException(e);
```

Comment:
handleCacheException will always throw, so how about we have the return type of the method be a runtime exception and then this becomes `throw unwrapCacheException(e)` (I also renamed the method)